mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -118,11 +118,14 @@ impl PeerManager {
         let mut peer_states = self.peer_states.lock();
         let is_new_peer = !peer_states.contains_key(&peer);
         if self.is_upstream_peer(&peer, Some(&metadata)) {
-            counters::active_upstream_peers(&peer.raw_network_id()).inc();
             // If we have a new peer, let's insert new data, otherwise, let's just update the current state
             if is_new_peer {
+                counters::active_upstream_peers(&peer.raw_network_id()).inc();
                 peer_states.insert(peer, PeerSyncState::new(metadata));
             } else if let Some(peer_state) = peer_states.get_mut(&peer) {
+                if !peer_state.is_alive {
```

Comment:
Shouldn't we decrement the alive peers if we're still going to add them again (when they are alive)? If not, we can still get double counts, right? i.e., add peer x (active peers = 1), remove peer X (which disables it, active peers = 1), add peer x (which enables it here, but re-increments, which means active peers = 2, but it's still only 1 peer?)

Reply:
Yeah, I realize now it never decremented.  Fixed now